roles/ceph-rgw/tasks/multisite/secondary.yml

```diff
@@ -1,28 +1,25 @@
 ---
-- name: ensure connection to primary cluster from mon
-  uri:
-    url: "{{ item.endpoint }}"
-  delegate_to: "{{ groups[mon_group_name][0] }}"
-  run_once: true
-  loop: "{{ secondary_realms }}"
-  when: secondary_realms is defined
+- name: create list secondary_realms
+  set_fact:
+    secondary_realms: "{{ secondary_realms | default([]) | union([{ 'realm': item.rgw_realm, 'zonegroup': item.rgw_zonegroup, 'zone': item.rgw_zone, 'endpoint': item.endpoint, 'system_access_key': item.system_access_key, 'system_secret_key': item.system_secret_key, 'is_master': item.rgw_zonemaster | default(rgw_zonemaster) }]) }}"
+  loop: "{{ rgw_instances }}"
+  when: not item.rgw_zonemaster | default(rgw_zonemaster) | bool
 
 - name: ensure connection to primary cluster from rgw
   uri:
     url: "{{ item.endpoint }}"
-  loop: "{{ rgw_instances }}"
+  loop: "{{ secondary_realms }}"
+  when: secondary_realms is defined
```

Comment:
Why are you removing the uri task from the monitor nodes ?

As far as I remember we needed both [1]

https://github.com/ceph/ceph-ansible/pull/5797#discussion_r490345434